src/handlers/platformProviderResponsesHandlers.ts

```diff
@@ -0,0 +1,124 @@
+import { ajv } from '../ajv';
+import { db } from '../database';
+import { isTinyPgErrorWithQueryContext } from '../types';
+import {
+  DatabaseError,
+  InputValidationError,
+} from '../errors';
+import type {
+  Request,
+  Response,
+  NextFunction,
+} from 'express';
+import type { JSONSchemaType } from 'ajv';
+import type { PlatformProviderResponse } from '../types';
+
+interface GetPlatformProviderResponseByExternalIdParams {
+  externalId: string;
+}
+const getPlatformProviderResponsesByExternalIdParamsSchema
+: JSONSchemaType<GetPlatformProviderResponseByExternalIdParams> = {
+  type: 'object',
+  properties: {
+    externalId: {
+      type: 'string',
+      pattern: '^\\d{9}$',
+    },
+  },
+  required: [
+    'externalId',
+  ],
```

Comment:
Can the read API also require a `provider` parameter, so that the data viewer can request something like `/platformProviderResponses?externalId=12-3456789&provider=CharityNavigator`?

Reply:
I think supporting a provider is a fine idea; requiring it is overly prescriptive to me.  What happens if you wanted to avoid two API calls?  What happens when there are 5 providers?

I realize this is a bespoke API endpoint, but I admit I assumed you'd want all of the provider responses in one swoop / you'd parse that single response, in the name of network efficiency.

Reply:
> I assumed you'd want all of the provider responses in one swoop / you'd parse that single response, in the name of network efficiency (at the very least it seems like it'd be good for the API to support either approach).

Oh! Yes, I think you're right. I was initially worried about typing the response, because although the data-scripts and this repo don't need to care about the shape of the data platform providers' responses, the data-viewer does, but I don't think that's actually a significant difference in implementation.